src/python/grpcio/grpc/_cython/_cygrpc/grpc_gevent.pyx

```diff
@@ -0,0 +1,437 @@
+# Copyright 2018 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# distutils: language=c++
+
+cimport cpython
+from libc cimport string
+from libc.stdlib cimport malloc, free
+import errno
+gevent_g = None
+gevent_socket = None
+gevent_hub = None
+gevent_event = None
+
+g_event = None
+g_greenlets = None
+
+cdef grpc_socket_vtable gevent_socket_vtable
+cdef grpc_custom_resolver_vtable gevent_resolver_vtable
+cdef grpc_custom_timer_vtable gevent_timer_vtable
+cdef grpc_custom_poller_vtable gevent_pollset_vtable
+
+def init_grpc_gevent():
+  _initialize_grpc_gevent_loop(False)
+
+def initialize_grpc_gevent_loop_test():
```

Comment:
Are we mixing production code and test code? How hard did you look at keeping them separate? Should we file a low-priority cleanup issue documenting that we're mixing production code and test code?

Reply:
The test code was a vestige of running the c-core end-to-end tests with gevent.

Because that is not part of this PR, I've removed it for now.